packages/vanilla/src/util/index.ts

```diff
@@ -153,6 +153,10 @@ export const vanillaStyles = [
     name: 'horizontal.layout',
     classNames: ['horizontal-layout']
   },
+  {
+    name: 'horizontal.layout.item',
+    classNames: numberOfChildren => ['horizontal-layout-' + numberOfChildren[0]]
```

Comment:
I think our typings are broken here right now, because `numberOfChildren` is a nested array ATM. We can fix this the following way (which also reflects the intended usage).
`StyleDef` should be defined this way (note varargs siganture on `args`):
```ts
export interface StyleDef {
  name: string;
  classNames: string[] | ((...args: any[]) => string[]);
}
```
If `classNames` is a function, we need to spread the `args` as well (see `styles.ts`):
```ts
    return foundStyle.classNames(...args);
```
This way, a `classNames` function can be used the following way (can we also use this as the example?):
```ts
{
    name: 'horizontal.layout.item',
    classNames: ([size, somethingElse]) => {
      const cols = 12 / size;
      console.log(somethingElse);
      return [`col-md-${cols}`];
    }
  }
```